src/nft/components/NFTMintCard.tsx

```diff
@@ -0,0 +1,58 @@
+import { background, border, cn, color } from '../../styles/theme';
+import { useIsMounted } from '../../useIsMounted';
+import { useTheme } from '../../useTheme';
+import { LifecycleType, type NFTMintCardReact } from '../types';
+import { NFTLifecycleProvider } from './NFTLifecycleProvider';
+import { NFTProvider } from './NFTProvider';
+
+export function NFTMintCard({
+  children,
+  className,
+  contractAddress,
+  tokenId,
+  useNFTData,
+  buildMintTransaction,
+  onStatus,
+  onError,
+  onSuccess,
+}: NFTMintCardReact) {
+  const componentTheme = useTheme();
+
+  const isMounted = useIsMounted();
+
+  // prevents SSR hydration issue
+  if (!isMounted) {
+    return null;
+  }
```

Comment:
i'm not familar with the order of these components but do we need the `isMounted` in both this component and `NFTCard`?